### ISSUE TITLE
Running the program

### ISSUE THREAD

**Issue by XKoop7321** [anakrusis/SMBMusEdit#1]
I'm trying to run this program, I have the SMB1 rom and your code, but how do I run the gui or the program itself? Thank you,

**Comment by repusoiram** [comment]
@Xkoop7321 I can help you. Let's have a conversation here so everyone else who wants to use this amazing program can follow along. Admittedly, it was very difficult to set this up as I'm new to this type of thing, but after hours of research and studying, I was able to get it to work. First off, you need Lua, version 5.1.5 to get this to work. Hopefully it's OK to post this link: https://sourceforge.net/projects/luabinaries/files/5.1.5/

you'll also need LOVE:
https://love2d.org/

also need utf8.lua :
https://gist.github.com/Stepets/3b4dbaf5e6e6a60f3862#file-utf8-lua

To download files from gethub, click EVERY file one at a time and download them to a folder, assets go in an asset folder within this program's filepath. If you see a file that you can't download, like main.lua. you'll need to create a text file, paste in the code, and change the file extension to .lua

with the above knowledge, google, and asking question here, we'll get you up and running. This alone will save you the hours I spent trying to get this to work.

There are other steps such as using the command prompt and setting environment variables.

Shout out to anakrusis who obviously put A LOT of time into this code. Terrific job! Probably saved me weeks of figuring out how to do this manually using the SMB1 music guide in romhacking.net. Thank you!!!!

-repuSoiraM

**Comment by XKoop7321** [comment]
Oh ok thank you but 1 question, will this work with smb rom hacks?


-Xkoop7321

**Comment by XKoop7321** [comment]
Say could you make a youtube video for this? Thank you!

**Comment by repusoiram** [comment]
yes it works for SMB1 Roms... you'll need the original US version (not the duck hunt / Track version). Sorry, I don't have time to make a YT vid but I can monitor and check this thread sporadically.

**Comment by anakrusis** [comment]
Hi, I apologise for not seeing this earlier, I haven't been on github much these past few months. There will be a built version that can run as an executable soon, but in the meantime it can still be run from LOVE directly, as repusoiram described (thank you very much for your comment, it indeed is helpful!)

A few things to add:
I wrote this for LOVE 11.3, when installed to a folder, it should include all the necessary lua dlls, and it will look like this to begin:
![image](https://user-images.githubusercontent.com/26858560/183345819-2ed0a270-86e9-4290-99e7-f9d432b63833.png)

then you can git clone smbmusedit directly into this folder, and it should appear as a folder within the folder:
![image](https://user-images.githubusercontent.com/26858560/183345962-89182fc3-8b22-4bd1-b408-2a00db8f45ef.png)

And finally, you just drag-and-drop the SMBMusEdit folder onto the love.exe and it should run!

**Comment by XKoop7321** [comment]
oh wait hold on I thought I WAS talking to you lol. So I just need the love program? or do I need all of them

**Comment by XKoop7321** [comment]
also do you think you could make a famitracker addon? I have a song I wanted to import into smb1 from famitracker without the hard way. if you could, that would be great!

**Comment by anakrusis** [comment]
Just installing LOVE should be enough, it should include all the lua stuff in it

Famitracker addon would be really challenging! because the SMB1 sound engine is very limited and can't handle almost any famitracker modules if they were to be imported into the game. So the .ftm would have to be made with all limitations in mind: very limited envelopes, limited pitch ranges, only eight possible note lengths, etc.-- and most importantly the very tight space restrictions!
I do decently understand the ftm file format though, so maybe some time in the future it could be attempted

**Comment by XKoop7321** [comment]
Ok. You can do anything if you put your mind to it!

**Comment by smfhacking** [comment]
@anakrusis Hello, sorry, I tried to drag SMBmusEdit into LOVE but this appears, what do I do?

if you could help me I would appreciate it very much
![Captura](https://github.com/anakrusis/SMBMusEdit/assets/138112106/3ae9909f-66a0-4fdd-9b97-6244210362f0)

**Comment by XKoop7321** [comment]
Seems to be a video card error. You might need to look for drivers which support OpenGK 2.1 or OpenGL ES 2 like it says. Your drivers might need to be updated as well.

**Comment by raz-w** [comment]
this is a great tool anakrusis, thank you
